src/app/context/theme.context.tsx

```diff
@@ -0,0 +1,30 @@
+'use client';
+
+import React, { Dispatch, ReactNode, SetStateAction, createContext, useContext, useMemo, useState } from 'react';
+import cn from 'classnames';
+
+interface TTheme {
+  theme: 'light' | 'dark';
+}
+
+interface TThemeContext extends TTheme {
+  setTheme: Dispatch<SetStateAction<'light' | 'dark'>>;
```

Comment:
Why don’t you reuse TTheme here? What if I tell you to add one more theme like pink, will change in in several places instead of one?